src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/HtmlHelperOfT.cs

```diff
@@ -63,6 +63,28 @@ public override void Contextualize(ViewContext viewContext)
             }
 
             ViewData = viewContext.ViewData as ViewDataDictionary<TModel>;
+
+            if (ViewData == null)
+            {
+                // The view data that we have at this point might be of a more derived type than the one defined at compile time.
+                // For example ViewDataDictionary<Cat> where our TModel is animal and Cat : Animal.
+                // This can happen if someone modified the model type through the page application model.
+                // In that case, we check if the type of the current view data, 'ViewData<TRuntime>' is "covariant" with the
+                // one defined at compile time 'ViewData<TModel>'
+                // When a type T<K> is covariant it means that an instance of that type T<'a> can be safely assigned to a variable
```

Comment:
Super don't need this explanation here. Just say we're allowing this case `VDD<Base> base; VDD<Derived> derivied; where Derived inherits from Base. We'll allow this case base = derived;

Reply:
Agree we don't need all the words explaining covariant because that's what Bing is for. However, *do* need to explain that `ViewDataDictionary<Derived>` can't occur for regular views -- only pages.